radsat-sk/operation/subsystems/RPdb.c

```diff
@@ -0,0 +1,272 @@
+                                            DEFINITIONS
+***************************************************************************************************/
+/**
+ * Conversion Constants for between ADC Values and Real Values
+ */
+
+/** Used in: (Volts = Constant * ADC Count) */
+#define ADC_COUNT_TO_BCR_OUTPUT_VOLTAGE			((float) 0.008993157)
+#define ADC_COUNT_TO_BCR_VOLTAGE				((float) 0.0322581)
+#define ADC_COUNT_TO_BATTERY_BUS_OUTPUT_VOLTAGE	((float) 0.008978)
+#define ADC_COUNT_TO_5V_BUS_OUTPUT_VOLTAGE		((float) 0.005865)
+#define ADC_COUNT_TO_3V3_BUS_OUTPUT_VOLTAGE		((float) 0.004311)
+
+
+/** Used in: (mAmps = Constant * ADC Count) */
+#define ADC_COUNT_TO_BCR_OUTPUT_CURRENT			((float) 14.662757)
+
+/** Used in: (Amps = Constant * ADC Count) */
+#define ADC_COUNT_TO_BCR_AMPERAGE				((float) 0.0009775)
+#define ADC_COUNT_TO_3V3_PDB_CURRENT_DRAW		((float) 0.001327547)
+#define ADC_COUNT_TO_5V_PDB_CURRENT_DRAW		((float) 0.001327547)
+#define ADC_COUNT_TO_BATTERY_BUS_OUTPUT_CURRENT	((float) 0.005237)
+#define ADC_COUNT_TO_5V_BUS_OUTPUT_CURRENT		((float) 0.005237)
+#define ADC_COUNT_TO_3V3_BUS_OUTPUT_CURRENT		((float) 0.005237)
+
+
+/** Used in: (Temp in Deg C = (Scale * ADC Count) - Shift) */
+#define ADC_COUNT_TO_SOLAR_ARRAY_TEMP_SCALE	((float) 0.4963)
+// Can use this \/ interchangeably between solar array and motherboard temp shift
+#define ADC_COUNT_TO_MOTHERBOARD_TEMP_SHIFT ((float) 273.15)
+#define ADC_COUNT_TO_MOTHERBOARD_TEMP_SCALE ((float) 0.372434)
+
+
+/** Used in: (Irradiance = Constant * ADC Count) */
+#define ADC_COUNT_TO_IRRADIANCE				((float) 1.59725)
+
+
+/**
+ * Constant Values for I2C Communication
+ */
+/** I2C Slave Address for PDB */
+#define PDB_I2C_SLAVE_ADDR 			(0x2B)
+/** Data Sent to the PDB via I2C for telemetry commands is 6 Bytes, 2 for the command, 4 for data*/
+#define PDB_TELEM_COMMAND_LENGTH	(6)
+/** Data sent to the PDB via I2C for all other commands is 4 Bytes, 2 for the command, 2 for data*/
+#define PDB_COMMAND_LENGTH			(4)
+/** Data Returned from the PDB via I2C is 4 Bytes */
+#define PDB_RESPONSE_LENGTH			(4)
+/** Delay between telemetry read/write operations with the PDB on I2C is 5ms */
+#define PDB_I2C_TELEM_DELAY			(5)
+/** Delay between other read/write operations with the PDB on I2C is 1ms */
+#define PDB_I2C_DELAY				(1)
+
+
+/**
+ * Constant Values for I2C Communication
+ */
+/** Number of sun sensors on the CubeSat */
+#define NUM_SUN_SENSORS		(6)
+
+
+
+
+/***************************************************************************************************
+                                          PRIVATE GLOBALS
+***************************************************************************************************/
+/**
+ *  4 Byte commands to request data for each of the 6 sun-sensors
+ *	located on each face of the CubeSat Solar Arrays.
+ */
+static uint32_t PdbSunSensorCommandBytes[6] = {
+	0x10E11C,     // SA1A - xPos? TODO: Fill out which side is which connection
```

Comment:
I asked the mechanical team this, join the #mechanical chat in slack, and await their response

Reply:
@IsaacPoirier did this ever get confirmed for you?

Reply:
@tyrelkostyk It was! I worked with the mech team to decide what worked best (within the realm of usage of the board), and we came to a solution which has been implemented in my code and supplied to the Mech team so the solar panels are connected correctly to match